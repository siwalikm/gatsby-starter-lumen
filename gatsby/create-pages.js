'use strict';

const path = require('path');
const createCategoriesPages = require('./pagination/create-categories-pages.js');
const createTagsPages = require('./pagination/create-tags-pages.js');
const createPostsPages = require('./pagination/create-posts-pages.js');

const createPages = async ({ graphql, actions }) => {
  const { createPage } = actions;

  // 404
  createPage({
    path: '/404',
    component: path.resolve('./src/templates/not-found-template.js')
  });

  // Tags list
  createPage({
    path: '/tags',
    component: path.resolve('./src/templates/tags-list-template.js')
  });

  // Categories list
  createPage({
    path: '/categories',
    component: path.resolve('./src/templates/categories-list-template.js')
  });

  // Posts and pages from markdown
  const result = await graphql(`
    {
      allMarkdownRemark(
        filter: {frontmatter: {draft: {ne: true}}}, sort: {fields: frontmatter___date, order: DESC}
      ) {
        edges {
          node {
            frontmatter {
              template
              title
            }
            fields {
              slug
            }
          }
        }
      }
    }
  `);

  const { edges } = result.data.allMarkdownRemark;
  const pages = edges.filter((edge) => edge.node.frontmatter.template === 'page');
  const posts = edges.filter((edge) => edge.node.frontmatter.template === 'post');
  console.log(pages, posts);
  pages.forEach((edge) => {
    createPage({
      path: edge.node.fields.slug,
      component: path.resolve('./src/templates/page-template.js'),
      context: { slug: edge.node.fields.slug }
    });
  });

  posts.forEach((edge, index) => {
    createPage({
      path: edge.node.fields.slug,
      component: path.resolve('./src/templates/post-template.js'),
      context: {
        slug: edge.node.fields.slug,
        prevPost: index === posts.length - 1 ? null : posts[index + 1],
        nextPost: index === 0 ? null : posts[index - 1],
      }
    });
  });

  // Feeds
  await createTagsPages(graphql, actions);
  await createCategoriesPages(graphql, actions);
  await createPostsPages(graphql, actions);
};

module.exports = createPages;
