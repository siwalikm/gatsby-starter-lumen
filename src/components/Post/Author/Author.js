// @flow strict
import React from 'react';
import { getContactHref } from '../../../utils';
import styles from './Author.module.scss';
import { useSiteMetadata } from '../../../hooks';

type Props = {
  date: string,
  postSlug: string,
};

const Author = ({ date, postSlug }: Props) => {
  const { author } = useSiteMetadata();
  const postURL = `https://github.com/siwalikm/diary/blob/master/content${postSlug}.md`;
  const formattedDate = new Date(date).toLocaleDateString('en-US', {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  });
  const dateString = formattedDate ? `on ${formattedDate}` : '';

  return (
    <div className={styles['author']}>
      <p className={styles['author__bio']}>
        Published {dateString} by{' '}
        <a
          // className={styles['author__bio-twitter']}
          href={getContactHref('twitter', author.contacts.twitter)}
          rel="noopener noreferrer"
          target="_blank"
        >
          <strong>{author.name}</strong>
        </a>
        . You can{' '}
        <a
          // className={styles['author__bio-twitter']}
          href={postURL}
          rel="noopener noreferrer"
          target="_blank"
        >
          <strong>edit this page</strong>
        </a>{' '}
        here.
      </p>
    </div>
  );
};

export default Author;
