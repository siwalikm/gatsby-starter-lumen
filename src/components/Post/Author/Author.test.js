// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Author from './Author';
import siteMetadata from '../../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../../types';

describe('Author', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => render(siteMetadata),
      useStaticQuery.mockReturnValue(siteMetadata),
    );
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Author date={new Date().toString()} postSlug="" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
