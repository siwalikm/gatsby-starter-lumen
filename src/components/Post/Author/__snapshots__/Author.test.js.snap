// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author renders correctly 1`] = `
<div
  className="author"
>
  <p
    className="author__bio"
  >
    Published 
    on Aug 28, 2022
     by
     
    <a
      href="https://www.twitter.com/#"
      rel="noopener noreferrer"
      target="_blank"
    >
      <strong>
        Test name
      </strong>
    </a>
    . You can
     
    <a
      href="https://github.com/siwalikm/diary/blob/master/content.md"
      rel="noopener noreferrer"
      target="_blank"
    >
      <strong>
        edit this page
      </strong>
    </a>
     
    here.
  </p>
</div>
`;
